src/System.Text.Json/src/System/Text/Json/Reader/Utf8JsonReader.TryGet.cs

```diff
@@ -470,5 +493,43 @@ public bool TryGetDateTimeOffset(out DateTimeOffset value)
             ReadOnlySpan<byte> span = HasValueSequence ? ValueSequence.ToArray() : ValueSpan;
             return JsonHelpers.TryParseAsISO(span, out value, out int bytesConsumed) && span.Length == bytesConsumed;
         }
+
+        /// <summary>
+        /// Reads the next JSON token value from the source and parses it to a <see cref="Guid"/>.
+        /// Returns <see langword="true"/> if the entire UTF-8 encoded token value can be successfully
+        /// parsed to a <see cref="Guid"/> value.
+        /// Returns <see langword="false"/> otherwise.
+        /// </summary>
+        /// <exception cref="InvalidOperationException">
+        /// Thrown if trying to get the value of a JSON token that is not a <see cref="JsonTokenType.String"/>.
+        /// <seealso cref="TokenType" />
+        /// </exception>
+        public bool TryGetGuid(out Guid value)
+        {
+            if (TokenType != JsonTokenType.String)
+            {
+                throw ThrowHelper.GetInvalidOperationException_ExpectedString(TokenType);
+            }
+
+            ReadOnlySpan<byte> span = HasValueSequence ? ValueSequence.ToArray() : ValueSpan;
```

Comment:
I see that the DateTime is already doing a ToArray for ValueSequence; but it seems like in both cases it would be better to rent an array, copy, clear, and return.

For GUID, since you only support the 36 character form you could probably get away with stackalloc `6*36` (maximally escaped) and copy the sequence into that.